File: backend/models/INetwork.py
from typing import Callable, List
from pydantic import BaseModel

class IWire(BaseModel):
    weight: float
    errorDelta: float
    accErrorDelta: float
    regularisation: Callable

class INeuron(BaseModel):
    bias: float
    inputs: List[IWire]
    outputs: List[IWire]
    inputVal: float
    outputVal: float
    intputDelta: float
    outputDelta: float
    accDelta: float
    numAccumulatedDelta: float
    update: Callable

class IArchitecture(BaseModel):
    networkShape: List[int]
    activation: str
    outputActivation: str
    regularisation: str
    initZero: bool

class ITraining(BaseModel):
    batchSize: int
    learningRate: float
    regLambda: float
    lossFn: str

class IBuildParams(BaseModel):
    architectureParams: IArchitecture
    trainingParams: ITraining

class IRunParams(BaseModel):
    serviceID: str
    dataspace: str

File: src/app/types/INetwork.ts
export type IArchitecture = {
  networkShape: number[];
  activation: string;
  outputActivation: string;
  regularisation: string;
  initZero: boolean;
}

export type ITraining = {
  batchSize: number;
  learningRate: number;
  regLambda: number;
  lossFn: string;
}

export type IBuildParams = {
  architectureParams: IArchitecture;
  trainingParams: ITraining;
}

export type IRunParams = {
  serviceID: string;
  dataspace: string;
}

export type IBuildRes = {
  serviceID: string;
}

export type IRunRes = {
  epochs: number;
  loss: number;
}

export type IShapeParams = {
  inputShape: number;
  hiddenShape: number[];
  outputShape: number;
}